module-bsp/board/rt1051/bsp/magnetometer/magnetometer.cpp

```diff
@@ -294,37 +296,55 @@ namespace bsp
             return read == 1;
         }
 
+        bool sliderChangedPosition(const Measurements &current)
+        {
+            /// The magnetometer is quite noisy. In some cases, noise can switch slider mode.
+            /// So we did a few measurements to calculate the level of noises and it should work fine.
+            constexpr auto maxNoiseLevel = 45;
```

Comment:
consider moving the value to the ~74 line, where you have another definition :)

Is the noise level somehow calculated, or it's based on all dev's test values?

Reply:
Sure, you're right.

I've got logs from 4 phones and calculated the average. Now I don't see any other solution. The frequency of polling is very low so we aren't able to filter this data.

Reply:
ALS31300 Datasheet, page 16, you can change the sampling rate and enable filtering in the chip itself - maybe it will improve the noise floor ;)